File: partition-array-according-to-given-pivot.ts
{
    function pivotArray(nums: number[], pivot: number): number[] {
        const start = [];
        const middle = [];
        const end = [];

        for (const num of nums) {
            if (num === pivot) {
                middle.push(num);
                continue;
            }
            
            if (num > pivot)
                end.push(num)
            else 
                start.push(num)
        }

        return [...start, ...middle, ...end];
    };

    console.log(pivotArray([9,12,5,10,14,3,10], 10))
}

File: partition-array-according-to-given-pivot.rs
pub fn pivot_array(nums: Vec<i32>, pivot: i32) -> Vec<i32> {
    nums.iter()
        .copied()
        .filter(|&x| x < pivot)
        .chain(nums.iter().copied().filter(|&y| y == pivot))
        .chain(nums.iter().copied().filter(|&z| z > pivot))
        .collect()
}
